homeassistant/components/wsdot/config_flow.py

```diff
@@ -0,0 +1,157 @@
+
+        return self.async_show_form(
+            step_id=SOURCE_USER,
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_API_KEY): str,
+                }
+            ),
+            errors=errors,
+        )
+
+    async def async_step_import(self, import_info: dict[str, Any]) -> ConfigFlowResult:
+        """Handle a flow initialized by import."""
+        await self.async_set_unique_id(import_info[CONF_API_KEY])
+        self._async_abort_entries_match({CONF_API_KEY: import_info[CONF_API_KEY]})
+        wsdot_travel_times = wsdot_api.WsdotTravelTimes(import_info[CONF_API_KEY])
+        try:
+            travel_time_routes = await wsdot_travel_times.get_all_travel_times()
+        except wsdot_api.WsdotTravelError as ws_error:
+            if ws_error.status == 400:
+                return self.async_abort(reason="invalid_api_key")
+            return self.async_abort(reason="cannot_connect")
+
+        subentries = []
+        for route in import_info[CONF_TRAVEL_TIMES]:
+            maybe_travel_time = [
+                tt
+                for tt in travel_time_routes
+                # old platform configs could store the id as either a str or an int
+                if str(tt.TravelTimeID) == str(route[CONF_ID])
+            ]
+            if not maybe_travel_time:
+                return self.async_abort(reason="invalid_travel_time_id")
+            travel_time = maybe_travel_time[0]
+            route_name = travel_time.Name
+            unique_id = "_".join(travel_time.Name.split())
+            subentries.append(
+                ConfigSubentry(
+                    subentry_type=SUBENTRY_TRAVEL_TIMES,
+                    unique_id=unique_id,
+                    title=route_name,
+                    data=MappingProxyType(
+                        {CONF_NAME: travel_time.Name, CONF_ID: travel_time.TravelTimeID}
+                    ),
+                ).as_dict()
+            )
+
+        return self.async_create_entry(
+            title=DOMAIN,
+            data={
+                CONF_API_KEY: import_info[CONF_API_KEY],
+            },
+            subentries=subentries,
+        )
+
+    @classmethod
+    @callback
+    def async_get_supported_subentry_types(
+        cls, config_entry: ConfigEntry
+    ) -> dict[str, type[ConfigSubentryFlow]]:
+        """Return subentries supported by wsdot."""
+        return {SUBENTRY_TRAVEL_TIMES: TravelTimeSubentryFlowHandler}
+
+
+class TravelTimeSubentryFlowHandler(ConfigSubentryFlow):
+    """Handle subentry flow for adding WSDOT Travel Times."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> SubentryFlowResult:
+        """Add a new Travel Time subentry."""
+        client = self._get_entry().runtime_data
+        if client is None:
+            return self.async_abort(reason="no_travel_time")
+        travel_times = await client.wsdot_travel_times.get_all_travel_times()
```

Comment:
When fetching this data I would just store it in this object as a dict with the TravelTimeID as key and Name as value. This way you can just do a lookup instead of 1. fetching all travel times again 2. iterating over all again